tests/components/bosch_alarm/test_services.py

```diff
@@ -0,0 +1,152 @@
+    await setup_integration(hass, mock_config_entry)
+    with pytest.raises(ServiceValidationError):
+        await hass.services.async_call(
+            DOMAIN,
+            SET_DATE_TIME_SERVICE_NAME,
+            {
+                ATTR_CONFIG_ENTRY_ID: "bad-config_id",
+                DATETIME_ATTR: dt_util.now(),
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_bad_params(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the service call is done with incorrect params."""
+    await setup_integration(hass, mock_config_entry)
+    with pytest.raises(vol.MultipleInvalid):
+        await hass.services.async_call(
+            DOMAIN,
+            SET_DATE_TIME_SERVICE_NAME,
+            {
+                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+                DATETIME_ATTR: "",
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_bad_year(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the panel fails the service call."""
+    await setup_integration(hass, mock_config_entry)
+    mock_panel.set_panel_date.side_effect = ValueError()
+    with pytest.raises(ServiceValidationError):
+        await hass.services.async_call(
+            DOMAIN,
+            SET_DATE_TIME_SERVICE_NAME,
+            {
+                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+                DATETIME_ATTR: dt_util.now(),
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_connection_error(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the panel fails the service call."""
+    await setup_integration(hass, mock_config_entry)
+    mock_panel.set_panel_date.side_effect = asyncio.InvalidStateError()
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(
+            DOMAIN,
+            SET_DATE_TIME_SERVICE_NAME,
+            {
+                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+                DATETIME_ATTR: dt_util.now(),
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_unloaded(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the config entry is unloaded."""
+    await setup_integration(hass, mock_config_entry)
+    async with mock_config_entry.setup_lock:
+        await mock_config_entry.async_unload(hass)
```

Comment:
You can just request the mock_config_entry and do `mock_config_entry.add_to_hass` and you're set with an unloaded entry